lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -165,3 +165,30 @@ def get_collection_dependencies(self, collection_candidate):
             get_collection_version_metadata(collection_candidate).
             dependencies
         )
+
+    def get_signatures(self, fqcn, version):
```

Comment:
Seems like `fqcn` and `version` are tightly coupled. Why not accept a `Constraint` or a `Requirement`?

Also, is it possible that different Galaxy / AH instances have different signatures? In this case, it'd be reasonable to extract the desired galaxy instance from such a requirement instance, like it's done in other methods above.

Another reason for doing this is consistency with other API methods.

Reply:
P.S. Having function annotations would be really helpful to understand what this method returns.

Reply:
+1, updated to take a Candidate argument and added type annotations